core/src/main/scala-2/net/ruippeixotog/akka/testkit/specs2/impl/CompatMatchers.scala

```diff
@@ -0,0 +1,27 @@
+package net.ruippeixotog.akka.testkit.specs2.impl
+
+import org.specs2.matcher.Expectable
+import org.specs2.matcher.Matchers._
+import org.specs2.matcher.MatchResult
+import org.specs2.matcher.StandardMatchResults.ok
+import org.specs2.matcher.ValueCheck
+
+import net.ruippeixotog.akka.testkit.specs2.Util._
+import net.ruippeixotog.akka.testkit.specs2.api.ReceiveMatcher
+import net.ruippeixotog.akka.testkit.specs2.impl.Matchers.TimeoutFunc
+import net.ruippeixotog.akka.testkit.specs2.impl.Matchers.GetMessageFunc
+
+private[specs2] object CompatMatchers {
+  abstract class ReceiveMatcherImpl[P, A](implicit tf: TimeoutFunc[P]) extends ReceiveMatcher[P, A] {
```

Comment:
Can we do even better here? I'd rather extract only specs2-specific primitive utilities (`result` functions, `mapCheck`) and leave akka logic to the appropriate implementation classes.

Reply:
Yes, take a look at https://github.com/ruippeixotog/akka-testkit-specs2/pull/282/commits/79c8daf3f942f8820f99a1194e9162fba5a0f6d2 🙂